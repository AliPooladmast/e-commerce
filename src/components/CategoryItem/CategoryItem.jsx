import React from "react";

const CategoryItem = () => {
  return <div>categoryItem</div>;
};

export default CategoryItem;
