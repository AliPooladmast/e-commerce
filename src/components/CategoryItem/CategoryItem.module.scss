.Container {
  flex: 1;
  margin: 0 20px;
  height: 60vh;
  position: relative;
  box-shadow: 0px 0px 5px 1px rgba(0, 0, 0, 0.3);

  img {
    width: 100%;
    height: 100%;
    object-fit: cover;

    @media only screen and (max-width: 600px) {
      height: 50vh;
    }
  }
}

.Info {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  h1 {
    background-color: rgba(255, 255, 255, 0.7);
    padding: 5px 10px;
    color: darkblue;
    font-size: 24px;
    border: 1px solid grey;
    box-shadow: 0px 0px 5px 1px rgba(0, 0, 0, 0.3);

    &:hover {
      background-color: rgba(255, 255, 255, 0.5);
    }
  }
}
