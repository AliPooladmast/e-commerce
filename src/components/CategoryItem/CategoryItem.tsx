import React from "react";
import { Link } from "react-router-dom";
import style from "./CategoryItem.module.scss";

enum categoryEnum {
  casual,
  fashion,
  formal,
}

interface IItem {
  id: number;
  img: string;
  imgMobile: string;
  title: string;
  category: categoryEnum;
}

const CategoryItem = ({ item }: { item: IItem }) => {
  return (
    <div className={style.Container}>
      <img
        src={item.img}
        srcSet={`${item.imgMobile} 200w, ${item.img} 500w`}
        sizes="(max-width: 380px) 200px, 500px"
        alt={item.title}
      />
      <div className={style.Info}>
        <Link to={`products/${item.category}`} className={style.Link}>
          <h1 data-testid="itemTitle">{item.title}</h1>
        </Link>
      </div>
    </div>
  );
};

export default CategoryItem;
