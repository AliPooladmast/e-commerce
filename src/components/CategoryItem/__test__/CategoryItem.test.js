import React from "react";
import { createRoot } from "react-dom/client";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import CategoryItem from "../CategoryItem";

const item = {
  img: "./assets/images/casual-dress.jpg",
  imgMobile: "./assets/images/casual-dress-mobile.jpg",
  title: "Casual Outfit",
  category: "casual",
};

it("renders without crashing", () => {
  const div = document.createElement("div");
  const root = createRoot(div);
  root.render(<CategoryItem />);
});

it("renders categoryItem correctly", () => {
  const { getByTestId } = render(
    <MemoryRouter>
      <CategoryItem item={item} />
    </MemoryRouter>
  );

  expect(getByTestId("itemTitle")).toHaveTextContent("Casual Outfit");
});

it("snapshot", () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <CategoryItem item={item} />
      </MemoryRouter>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
