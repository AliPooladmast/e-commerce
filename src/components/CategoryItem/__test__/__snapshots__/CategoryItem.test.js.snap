// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="Container"
>
  <img
    alt="Casual Outfit"
    sizes="(max-width: 380px) 200px, 500px"
    src="./assets/images/casual-dress.jpg"
    srcSet="./assets/images/casual-dress-mobile.jpg 200w, ./assets/images/casual-dress.jpg 500w"
  />
  <div
    className="Info"
  >
    <a
      className="Link"
      href="/products/casual"
      onClick={[Function]}
    >
      <h1
        data-testid="itemTitle"
      >
        Casual Outfit
      </h1>
    </a>
  </div>
</div>
`;
